File: Web/src/api/category.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetCategoryList() {
  return apiClient.post("/category/GetCategoryList").then(handleResponse).catch(handleError)
}

function GetCategory(data) {
  return apiClient.post("/category/GetCategory", data).then(handleResponse).catch(handleError)
}

function CreateCategory(data) {
  return apiClient.post("/category/CreateCategory", data).then(handleResponse).catch(handleError)
}

function UpdateCategory(data) {
  return apiClient.post("/category/UpdateCategory", data).then(handleResponse).catch(handleError)
}

function DeleteCategory(data) {
  return apiClient.post("/category/DeleteCategory", data).then(handleResponse).catch(handleError)
}

const categoryApi = {
  GetCategoryList,
  GetCategory,
  CreateCategory,
  UpdateCategory,
  DeleteCategory
}

export default categoryApi

File: Server/route/category.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetCategoryRoute(r *gin.Engine) {
	category := r.Group("/category")
	category.Use(middleware.JWTAuthMiddleware())
	{
		category.POST("/GetCategoryList", api.GetCategoryList)
		category.POST("/GetCategory", api.GetCategory)
	}
	adminCategory := r.Group("/category")
	adminCategory.Use(middleware.AdminAuthMiddleware())
	{
		adminCategory.POST("/CreateCategory", api.CreateCategory)
		adminCategory.POST("/UpdateCategory", api.UpdateCategory)
		adminCategory.POST("/DeleteCategory", api.DeleteCategory)
	}
}
